service/src/main/kotlin/fi/espoo/evaka/attachment/AttachmentsController.kt

```diff
@@ -114,6 +115,30 @@ class AttachmentsController(
         return id
     }
 
+    data class PreDownloadResponse(
+        val fileAvailable: Boolean
+    )
+
+    @GetMapping("/{attachmentId}/pre-download")
+    fun checkAttachmentAvailability(
+        db: Database,
+        user: AuthenticatedUser,
+        @PathVariable attachmentId: UUID
+    ): ResponseEntity<PreDownloadResponse> {
+        Audit.AttachmentsRead.log(targetId = attachmentId)
+        if (!user.hasOneOfRoles(UserRole.ADMIN, UserRole.SERVICE_WORKER, UserRole.FINANCE_ADMIN)) {
+            if (!db.read { it.isOwnAttachment(attachmentId, user) }) throw Forbidden("Permission denied")
+        }
+
+        return try {
+            s3Client.headObject(filesBucket, "$attachmentId").let { _ ->
+                ResponseEntity.ok().body(PreDownloadResponse(true))
+            }
+        } catch (e: AmazonS3Exception) {
+            ResponseEntity.ok().body(PreDownloadResponse(false))
+        }
```

Comment:
I think it'd be better to explicitly only handle the forbidden case (looks like there isn't a specific class for that and might have to just check `.statusCode`). Currently we would swallow every other service exception (though not client exceptions, like connection errors) and it would make debugging harder.